sdk/ai/ai-inference-rest/src/tracingPolicy.ts

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { PipelinePolicy, PipelineRequest, PipelineResponse, SendRequest } from "@azure/core-rest-pipeline";
+import {
+  type TracingClient,
+  type TracingContext,
+  type TracingSpan,
+  createTracingClient,
+} from "@azure/core-tracing";
+
+
+import { getErrorMessage, isError } from "@azure/core-util";
+import { SDK_VERSION } from "./constants.js";
+import { logger } from "./logger.js";
+
+/**
+ * The programmatic identifier of the tracingPolicy.
+ */
+export const tracingPolicyName = "Inference_Tracing";
+
+const tracingClient = tryCreateTracingClient();
+
+/**
+ * Options to configure the tracing policy.
+ */
+export interface TracingPolicyOptions {
+  /**
+   * Query string names whose values will be logged when logging is enabled. By default no
+   * query string values are logged.
+   */
+  additionalAllowedQueryParameters?: string[];
+}
+
+
+export enum TracingAttributesEnum {
+  Operation_Name="gen_ai.operation.name",
+  Request_Model="gen_ai.request.model",
+  System="gen_ai.system",
+  Error_Type="error.type",
+  Server_Port="server.port", // not use it for now
+  Request_Frequency_Penalty="gen_ai.request.frequency_penalty",
+  Request_Max_Tokens="gen_ai.request.max_tokens",
+  Request_Presence_Penalty="gen_ai.request.presence_penalty",
+  Request_Stop_Sequences="gen_ai.request.stop_sequences",
+  Request_Temperature="gen_ai.request.temperature",
+  Request_Top_K="gen_ai.request.top_k",
+  Request_Top_P="gen_ai.request.top_p",
+  Response_Finish_Reasons="gen_ai.response.finish_reasons",
+  Response_Id="gen_ai.response.id",
+  Response_Model="gen_ai.response.model",
+  Usage_Input_Tokens="gen_ai.usage.input_tokens",
+  Usage_Output_Tokens="gen_ai.usage.output_tokens",
+  Server_Address="server.address"
+}
+
+export type TracingAttributes = { [key in TracingAttributesEnum]?: string | number};
+
+/**
+ * A simple policy to create OpenTelemetry Spans for each request made by the pipeline
+ * that has SpanOptions with a parent.
+ * Requests made without a parent Span will not be recorded.
+ */
+export function tracingPolicy(): PipelinePolicy {
+
+  return {
+    name: tracingPolicyName,
+    async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
```

Comment:
The API, post function is in the core-client-rest.  Not in ai-inference.   So Instead of using a tracing policy in the rest pipeline, I create a tracer within the inference library, and pass it to core-client-rest. While calling post or other method, the tracer will tap the implement and create span using the arguments and result.

Reply:
Done